### ISSUE TITLE
Download stuck on "Downloading required file"

### ISSUE THREAD

**Issue by Lythenas** [PrismLauncher/PrismLauncher#2431]
### Operating System

Linux

### Version of Prism Launcher

8.3

### Version of Qt

Qt 6.7.0
Also tried with the AppImage which had Qt 6.2.4

### Description of bug

When creating a new instance everything looks like it worked. But then when starting the instance it starts again downloading some things and then gets stuck. Eventually the list of things to download in the UI is empty and the progess bar slowly fills up. But it never completes.

In the logs I saw a bunch of lines like this:

```
   995.977 D | Setting up api download
  1025.976 C | [launcher.task.net.download]: "{ff3ac0e7-a34d-4c07-952b-d3f58dd9ee79}" Aborted  "https://maven.minecraftforge.net/org/jline/jline/3.5.1/jline-3.5.1.jar"
```

I tried this with a lot of different modpacks from basically all sources.

Launching vanilla minecraft works fine.

### Steps to reproduce

- Create a new instance of a modpack
- launch it
- see that "Downloading required library files" never completes

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by Trial97** [comment]
Try reducing the number of concurent downloads.
Try the nightly builds. 
And if nothing works a full log would help(upload it to a pastebin service)

**Comment by Lythenas** [comment]
TLDR The fix for me was to disable IPv6 on my network connection.

Setting the number of concurrent downloads to 1 helper a little bit but wasn't enough to finish downloading.

I didn't have any problems downloading the mods through the browser and it downloaded fast. But for some reason downloading in Prism Launcher was very slow (when it would even start).

I saw in the logs that it showed IPv4 and IPv6 addresses, so I tried to download using IPv4 and IPv6 with curl. And there was the issue. With IPv4 it was as fast as in the browser. But with IPv6, curl first took >5s to start the download and was then also really slow (like 3 KB/s). I don't know if this is an issue with https://maven.minecraftforge.net or with my network.

So the fix for me was to disable IPv6 on my network connection.

I don't know if there is something you can do to improve this situation in the Prism Launcher. Maybe give IPv4 precedence over IPv6 for minecraftforge.net URLs, if that is possible. Trying both IPv4 and 6 and picking the faster one seems hard to do.

Anyway since my problem is resolved, feel free to close this issue if you don't see a way to improve this in Prism Launcher.

**Comment by Trial97** [comment]
Sadly I looked this up and I could not make qt choose an interface for the requests(the os handles that).
So the best solution here is to disable the network interfaces through the os.
Gonna close this as there is no way to do it in QT.

**Comment by mo-db** [comment]
@Lythenas removing ipv6 fixed the issue for me too, thx!

**Comment by RevLaw** [comment]
Disabling ipv6 is still the solution in the newest version.
Is there a fix planned, like blocking ipv6 on the launcher site?

**Comment by NousadS** [comment]
I have ipv6 enabled and solve problem just by terminating the process (not closing, explicitly terminating through system monitor/terminal)

**Comment by hellyet** [comment]
Have ipv6 disabled and still facing this issue.

<img width="488" height="261" alt="Image" src="https://github.com/user-attachments/assets/0c0e2ff1-103c-4179-a014-7e29ae45d9d7" />

**Comment by Marqueee** [comment]
> Have ipv6 disabled and still facing this issue.
> <img alt="Image" width="488" height="261" src="https://private-user-images.githubusercontent.com/74832935/578726951-0c0e2ff1-103c-4179-a014-7e29ae45d9d7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzcwMTA4NTksIm5iZiI6MTc3NzAxMDU1OSwicGF0aCI6Ii83NDgzMjkzNS81Nzg3MjY5NTEtMGMwZTJmZjEtMTAzYy00MTc5LWEwMTQtN2UyOWFlNDVkOWQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA0MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNDI0VDA2MDIzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkYmY0NDcxOTdiNjdhMGMzNGNjODBmYzYwMjEwZDIzZTg5YTAyZWMwODVhZDRhMTdiY2I5N2ZhNTMwY2ZjNGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.ivkYZwm3LlvbgWdJw-tLH8fHr3cdevc_0r0D7e58WH0">

same trouble

**Comment by hellyet** [comment]
Network issue. Use proxy

**Comment by RevLaw** [comment]
For me the problem was im using a unifi dream machine router and the security setting was detecting Minecraft as "malicious" traffic. After disabeling the rule everything started working fine.

**Comment by hellyet** [comment]
UPD: Just re-installed my Windows and facing the same problem, but for now proxy doesn't fix anything - just can't download that damn library
@Trial97 is there anything we can do to fix this?
